packages/material-ui-lab/src/Slider/Slider.js

```diff
@@ -664,54 +757,58 @@ Slider.propTypes = {
    * Should not be equal to max.
    */
   min: PropTypes.number,
+  /**
+   * Name attribute of the hidden `input` element.
+   */
+  name: PropTypes.string,
   /**
    * Callback function that is fired when the slider's value changed.
+   *
+   * @param {object} event The event source of the callback
+   * @param {any} value The new value
    */
   onChange: PropTypes.func,
   /**
-   * Callback function that is fired when the slide has stopped moving.
+   * Callback function that is fired when the mouseup is triggerd.
+   *
+   * @param {object} event The event source of the callback
+   * @param {any} value The new value
    */
-  onDragEnd: PropTypes.func,
+  onChangeCommitted: PropTypes.func,
   /**
-   * Callback function that is fired when the slider has begun to move.
+   * @ignore
    */
-  onDragStart: PropTypes.func,
+  onMouseDown: PropTypes.func,
   /**
-   * The granularity the slider can step through values.
+   * The stepper orientation (layout flow direction).
    */
-  step: PropTypes.number,
+  orientation: PropTypes.oneOf(['horizontal', 'vertical']),
   /**
-   * @ignore
+   * The granularity the slider can step through values.
+   * When step is `null`, users can only slide the thumbs onto marks.
    */
-  theme: PropTypes.object.isRequired,
+  step: PropTypes.number,
   /**
-   * The component used for the slider icon.
-   * This is optional, if provided should be a react element.
+   * The thumb label.
    */
-  thumb: PropTypes.element,
+  ThumbLabelComponent: PropTypes.elementType,
   /**
-   * The value of the slider.
+   * Show thumb label.
    */
-  value: PropTypes.number.isRequired,
+  thumbLabelDisplay: PropTypes.oneOf(['on', 'active', 'off']),
```

Comment:
For consistency with the rest of the API, and with the spec, what you you think about making this a boolean? `false`, no value label, `true`, the value label is displayed on hover, focus, and click, per:

![image](https://user-images.githubusercontent.com/357702/59151926-11e9a200-8a33-11e9-9342-a92949fab2ab.png)

![image](https://user-images.githubusercontent.com/357702/59151939-2cbc1680-8a33-11e9-81b6-192e32885a9c.png)


Also, it seems the thumbLabel should be default for discreet, but not for ranged.

Reply:
What about the "on" state? I have followed the specification, up to an extent, I couldn't ignore the features of the other solutions I have benchmarked.

Reply:
I'm not sure to understand the consistency argument. We have precedence of this: `Tabs.scrollButtons`.

Reply:
Hmm. That's a shame. Well, I guess the precedent has been set. For slider I would have argued for using a style override for the non-standard use-case.

In any case, it should be `off` by default, at the least for continuous, and arguably `auto` by default for discrete. (I would be fine with `off` for both for consistency.

Also, `thumbLabel*` -> `valueLabel*`?

Reply:
The API was designed so there is not continuous/discrete difference. IMHO the whole concept is brittle. Ok, I have looked at a couple of alternatives, I think that it should be off by default 👍.